homeassistant/components/whirlpool/climate.py

```diff
@@ -0,0 +1,189 @@
+"""Platform for climate integration."""
+import asyncio
+import logging
+
+import aiohttp
+from whirlpool.aircon import Aircon, FanSpeed as AirconFanSpeed, Mode as AirconMode
+from whirlpool.auth import Auth
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from .const import AUTH_INSTANCE_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+AIRCON_MODE_MAP = {
+    AirconMode.Cool: HVAC_MODE_COOL,
+    AirconMode.Heat: HVAC_MODE_HEAT,
+    AirconMode.Fan: HVAC_MODE_FAN_ONLY,
+}
+
+HVAC_MODE_TO_AIRCON_MODE = {v: k for k, v in AIRCON_MODE_MAP.items()}
+
+AIRCON_FANSPEED_MAP = {
+    AirconFanSpeed.Off: FAN_OFF,
+    AirconFanSpeed.Auto: FAN_AUTO,
+    AirconFanSpeed.Low: FAN_LOW,
+    AirconFanSpeed.Medium: FAN_MEDIUM,
+    AirconFanSpeed.High: FAN_HIGH,
+}
+
+FAN_MODE_TO_AIRCON_FANSPEED = {v: k for k, v in AIRCON_FANSPEED_MAP.items()}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+    auth: Auth = hass.data[DOMAIN][config_entry.entry_id][AUTH_INSTANCE_KEY]
+    said_list = auth.get_said_list()
+    if not said_list:
+        _LOGGER.debug("No appliances found")
+        return
+
+    aircon = AirConEntity(said_list[0], auth)
+    async_add_entities([aircon], True)
```

Comment:
```suggestion
    async_add_entities([aircon])
```

Do we need to force an update when adding the entity?

Reply:
Yes, otherwise we will not get any values (the socket only sends changes)